### DISCUSSION TITLE
Add Text-to-Speech (SSML) with audio output

### DISCUSSION THREAD

**Discussion by SimonGolms** [WeblateOrg/weblate#5503] [Ideas]
I am currently developing a multilingual app and would like to use Weblate as a translation tool. As a further requirement of the accessibility of the app, a text-to-speech output should be possible, which could be implemented e.g. with the [Azure Cognitive Service / Speech Service / Text-To-Speech (TTS)](https://docs.microsoft.com/en-us/azure/cognitive-services/Speech-Service/text-to-speech) and the use of [Speech Synthesis Markup Language (SSML)](https://www.w3.org/TR/speech-synthesis/).

Ideally, I would like to have the SSML text building blocks translated by the contributing translators as well. Like already e.g. screenshots support the translation, I could imagine here e.g. a [play-button](https://material.io/resources/icons/?icon=play_arrow), which outputs the SSML input as audio for validation.

```json
// en-US.json
{
  "intro": "The SSML standard is defined by the W3C.",
  "intro_ssml": "The <say-as interpret-as=\"characters\">SSML</say-as> standard <break time=\"1s\"/>is defined by the <sub alias=\"World Wide Web Consortium\">W3C</sub>."
}
```

As far as I know, there is no free to use TTS service. However, the API of the common cloud providers could be integrated for this, similar to what is already the case for the Machine translating.

Via settings menu, the other parameters for a complete SSML can be configured:

```xml
<speak xmlns="http://www.w3.org/2001/10/synthesis" xmlns:mstts="http://www.w3.org/2001/mstts" xmlns:emo="http://www.w3.org/2009/10/emotionml" version="1.0" xml:lang="en-US">
  <voice name="en-US-AriaNeural">
    The <say-as interpret-as="characters">SSML</say-as> standard <break time="1s"/>is defined by the <sub alias="World Wide Web Consortium">W3C</sub>.
  </voice>
</speak>
```

**Comment by nijel** [comment]
I don't see a problem with integrating this. On the other side I don't think we will spend time developing something like this unless the development is funded.

At least two things should cover this:

* SSML markup checks so that input by translators is validated to be syntactically correct
* Integration with TTS services to allow playing the text

There is http://espeak.sourceforge.net/, but I have no clue how maintained it is or how easy it is to integrate with it.

**Reply by SimonGolms** [reply]
> I don't see a problem with integrating this. On the other side I don't think we will spend time developing something like this unless the development is funded.

Glad to hear it and a perfectly valid point as well.
Maybe there could be more interested parties in the future, especially when it comes to e.g. the development of multilingual skills/apps for smart speakers or any other [voice user interface (VUI)](https://en.wikipedia.org/wiki/Voice_user_interface)
